### PR TITLE
changed the variant of the button to show a filter is active

### LINKED ISSUE #6931: Log Update - filter
**Is your feature request related to a problem? Please describe.**
When a user has applied filter for log update, there is no indication that a filter is in place.

**Describe the solution you'd like**
[A clear and concise description of what you want to happen.
](https://care.coronasafe.in/facility/fa4942c3-dce3-477e-81b2-7c95cd18b5da/patient/b26e30ab-7557-42d5-a5fb-14f23c086964/consultation/fae6f90b-c8d9-4593-83ff-1918f7a9bafd)

Apply a filter for log update. You will see a shorter list of log updates. However, the user maynot realise it.

Add a small icon on top of "filter" to show that a filter is in place. 

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by shramanpaul**
## Proposed Changes

- Fixes #6931

@coronasafe/care-fe-code-reviewers @coronasafe/code-reviewers

## Merge Checklist

https://github.com/coronasafe/care_fe/assets/110323017/1a0353d3-bd12-41e1-a5b9-430020020d0c

@rithviknishad please have a review

### CODE REVIEW

**Review by rithviknishad — CHANGES_REQUESTED** [review]
I don't think this looks good as a layout shift happens. The filter button becomes smaller when the filter is applied.

Instead we could change the `Button`'s variant to "primary" when a filter is present and keep it "secondary" if not.

**Comment by shramanpaul** [comment]
Thanks for the review. I will be making the requested changes as soon as possible

**Comment by shramanpaul** [comment]
https://github.com/coronasafe/care_fe/assets/110323017/cc4defb4-2092-450d-899e-72097c9e5a63

made the requested changes please have a review @rithviknishad

File: src/Components/Facility/Consultations/DailyRoundsFilter.tsx [hunk 1]
```diff
@@ -33,14 +33,31 @@ export default function DailyRoundsFilter(props: Props) {
...
+              </ButtonV2>
+            </>
+          ) : (
+            <>
+              <ButtonV2 variant="primary" className="mr-5 border">
+                <CareIcon className="care-l-filter" />
+                {t("filter")}
+              </ButtonV2>
+            </>
+          )}
```
Comment (rithviknishad) [inline src/Components/Facility/Consultations/DailyRoundsFilter.tsx]: This code is not readable. You are constructing two different buttons based on the condition, when only the `variant` prop is changing.

Instead pass the boolean condition to `variant` alone.


```suggestion
          <ButtonV2 variant={isFilterApplied ? "primary" : "secondary" className="mr-5 border">
            <CareIcon className="care-l-filter" />
            {t("filter")}
          </ButtonV2>
```

File: src/Components/Facility/Consultations/DailyRoundsFilter.tsx [hunk 2]
```diff
@@ -33,14 +33,31 @@ export default function DailyRoundsFilter(props: Props) {
     errorClassName: "hidden",
   });
 
+  const countDefinedProperties = (obj) => {
+    return Object.values(obj).filter((val) => val !== undefined).length;
+  };
```
Comment (rithviknishad) [inline src/Components/Facility/Consultations/DailyRoundsFilter.tsx]: Why do an iterate and check for length? You can simply use the `Array.every`/`Array.some` method for this instead, which would be much more readable.

```suggestion
  const isFilterApplied = Object.values(filter ?? {}).some(Boolean);
```

**Review by rithviknishad — CHANGES_REQUESTED** [review]

File: src/Components/Facility/Consultations/DailyRoundsFilter.tsx (hunk 1 above)
Comment (shramanpaul) [reply src/Components/Facility/Consultations/DailyRoundsFilter.tsx]: Thanks for the review. Yes the code was not readable. I have made the suggested changes.

**Review by rithviknishad — APPROVED** [review]
Cleaner! LGTM

File: src/Components/Facility/Consultations/DailyRoundsFilter.tsx [hunk 3]
```diff
@@ -33,11 +33,18 @@ export default function DailyRoundsFilter(props: Props) {
     errorClassName: "hidden",
   });
 
+  const isFilterApplied = Object.values(filter ?? {}).some(
+    (val) => val !== undefined
+  );
+
   return (
     <div className="flex flex-row-reverse items-center gap-4 md:flex-row">
       <Popover className="relative ">
```
Comment (Ashesh3) [inline src/Components/Facility/Consultations/DailyRoundsFilter.tsx]:
```suggestion
      <Popover className="relative">
```

**Review by Ashesh3 — APPROVED** [review]

**Comment by nihal467** [comment]
LGTM

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**